menoh/tensorrt/Inference.cpp

```diff
@@ -0,0 +1,358 @@
+#include <chrono>
+#include <iostream>
+
+#include <menoh/array.hpp>
+#include <menoh/exception.hpp>
+#include <menoh/graph.hpp>
+#include <menoh/utility.hpp>
+
+#include <NvInfer.h>
+#include <cuda_runtime_api.h>
+
+#include <menoh/tensorrt/Exception.hpp>
+#include <menoh/tensorrt/Inference.hpp>
+
+using namespace nvinfer1;
+
+#define CHECK(status)                             \
+    do {                                          \
+        auto ret = (status);                      \
+        if(ret != 0) {                            \
+            std::cout << "Cuda failure: " << ret; \
+            abort();                              \
+        }                                         \
+    } while(0)
+
+class Logger : public nvinfer1::ILogger {
+public:
+    Logger() : Logger(Severity::kWARNING) {}
+
+    Logger(Severity severity) : reportableSeverity(severity) {}
+
+    void log(Severity severity, const char* msg) override {
+        // suppress messages with severity enum value greater than the
+        // reportable
+        if(severity > reportableSeverity)
+            return;
+
+        switch(severity) {
+            case Severity::kINTERNAL_ERROR:
+                std::cerr << "INTERNAL_ERROR: ";
+                break;
+            case Severity::kERROR:
+                std::cerr << "ERROR: ";
+                break;
+            case Severity::kWARNING:
+                std::cerr << "WARNING: ";
+                break;
+            case Severity::kINFO:
+                std::cerr << "INFO: ";
+                break;
+            default:
+                std::cerr << "UNKNOWN: ";
+                break;
+        }
+        std::cerr << msg << std::endl;
+    }
+
+    Severity reportableSeverity{Severity::kWARNING};
+};
+
+namespace menoh_impl {
+    namespace tensorrt_backend {
+#ifdef MENOH_ENABLE_TENSORRT_PROFILER
+        struct Profiler : public IProfiler {
+            const int TIMING_ITERATIONS = 1;
+
+            typedef std::pair<std::string, float> Record;
+            std::vector<Record> mProfile;
```

Comment:
Can't we use `std::map<std::string, float>` (or unordered version) instead of `std::vector<Record>` ?

Reply:
Thank you comment. I think so too. `std::unordered_map` seems suitable here.